src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -472,18 +472,31 @@ export class VSCodeNotebookController implements Disposable {
         let controller = this.controller;
         let kernelStarted = false;
         try {
-            kernel = await this.connectToKernel(doc, new DisplayOptions(false));
-            kernelStarted = true;
-            // If the controller changed, then ensure to create a new cell execution object.
-            if (kernel && kernel.controller.id !== controller.id) {
-                controller = kernel.controller;
-                exec = this.createCellExecutionIfNecessary(cell, kernel.controller);
-            }
-            currentContext = 'execution';
-            if (kernel.controller.id === this.id) {
+            // If we're dealing with live kernel connections, then get the kernel.
+            if (this.kernelConnection.kind === 'connectToLiveLocalKernel') {
```

Comment:
Seems weird that this would be special cased here? Why doesn't it behave just like remote live kernels?

Reply:
Basically we need a IKernel connection.
With remote kernels the controller has access to the sesion connection information, & in lower layers of the code we start a connection & execute against that.
With live kernels, the kernel connection information has the connection information as well, however it only needs the id of the kernel to be used. And when running code, all we need to do is fetch the real kernel & use that & that's whats happening here. This is the top most layer of the code & all we do here is fetch the kernel and rest of the code works as expected.